src/fides/api/schemas/connection_configuration/connection_oauth_config.py

```diff
@@ -0,0 +1,47 @@
+from typing import ClassVar, List, Optional
+from enum import Enum
+
+from pydantic import BaseModel, ConfigDict, Field
+
+from fides.api.schemas.base_class import NoValidationSchema
+
+
+class OAuthGrantType(str, Enum):
+    """OAuth2 grant types supported by the system"""
+
+    authorization_code = "authorization_code"
+    client_credentials = "client_credentials"
+    password = "password"
+    implicit = "implicit"
```

Comment:
If I understand this correctly,`BackendApplicationClient` would only support the `client_credentials` grant type? If so, can we remove the other grant types so we only accept the supported grant types in the API schemas.
```suggestion
class OAuthGrantType(str, Enum):
    """OAuth2 grant types supported by the system"""

    client_credentials = "client_credentials"
```
    
    
    
    
    
    
```